orientdb/src/main/scala/akka/stream/alpakka/orientdb/OrientDBSinkSettings.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.orientdb
+
+//#sink-settings
+final case class OrientDBSinkSettings(maxPartitionSize: Int = Runtime.getRuntime.availableProcessors(),
+                                      maxPoolSize: Int = -1,
+                                      maxRetry: Int = 1,
+                                      retryInterval: Int = 5000,
+                                      bufferSize: Int = 10,
+                                      dbUrl: String,
+                                      username: String,
+                                      password: String)
+//#sink-settings
+{
+  def this(dbUrl: String, username: String, password: String) =
+    this(Runtime.getRuntime.availableProcessors(), -1, -1, 5000, 10, dbUrl, username, password)
+
+  def withMaxPartitionSize(maxPartitionSize: Int): OrientDBSinkSettings =
+    new OrientDBSinkSettings(
```

Comment:
this is a case class. You can simply do `copy(maxPartitionSize = maxPartitionSize)` to do the same.

And the same to al the other withXXX methods.